iib/web/models.py

```diff
@@ -360,6 +370,20 @@ def add_state(self, state, state_reason):
         db.session.flush()
         self.request_state_id = request_state.id
 
+    def add_build_tag(self, name):
+        """
+        Add a RequestBuildTag associated with the current request.
+
+        :param str name: tag name
+        """
+        bt = db.session.query(RequestBuildTag).filter_by(name=name).first()
+        if not bt:
+            bt = RequestBuildTag(name=name, request_id=self.id)
+            db.session.add(bt)
+            db.session.flush()
+        else:
```

Comment:
This should probably be `if bt not in self.build_tags`

Reply:
no, it's not needed if you set request_id=self.id. You can check newly added tests.